include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -6069,6 +6089,42 @@ struct OrtApi {
 
   /// @}
 
+  /// \name OrtExternalInitializerInfo
```

Comment:
Given this is an opaque type can we do something to handle the special 'in-memory' external data more nicely? External user shouldn't need to be parsing for magic tags to indicate that.

Reply:
To clarify a bit more: this public API always returns real external file information. Even if ORT has loaded the external data (mmap), we'll return the original external file information.

Reply:
In that case:
- calling `ValueInfo_GetInitializerValue` returns an OrtValue, which is agnostic of origin. There are no API functions off OrtValue that allow querying whether this is an "external initializer".
- calling `ValueInfo_GetExternalInitializerInfo` returns a NULL `OrtExternalInitializerInfo` since the data was not on disk.